module3/homework/vectorOfSharedPointers/vectorFunctions.cpp

```diff
@@ -0,0 +1,46 @@
+#include "vectorFunctions.hpp"
+
+#include <iostream>
+
+std::vector<std::shared_ptr<int>> generate(int count) {
+    std::vector<std::shared_ptr<int>> resultVector{};
+
+    if (count < 0) {
+        return resultVector;
+    }
+
+    resultVector.reserve(count);
```

Comment:
You can specify a size of vector in constructor :)
```suggestion
    std::vector<std::shared_ptr<int>> resultVector(count);

    if (count < 0) {
        return resultVector;
    }

```

Reply:
Yes, but what about situation, when count is negative value? I do not think such value should be used in the vector constructor, that's why I do such check first.

Reply:
you are right :) , my fault :)

Reply:
No problem, thanks for review :)